source/docs/software/actuators/solenoids.rst

```diff
@@ -0,0 +1,102 @@
+Operating pneumatic cylinders
+=============================
+
+Solenoid control
+----------------
+
+FRC teams use solenoids to preform a variety of tasks, from shifting
+gearboxes to operating robot mechanisms. A solenoid is a valve used to
+electronically switch a pressurised air line "on" or "off". For more
+information on solenoids, see `this wikipedia
+article <https://en.wikipedia.org/wiki/Solenoid_valve>`__. Solenoids are
+controlled by a robot's Pneumatics Control Module, or PCM, which is in
+turn connected to the robot's RoboRIO via CAN. The easiest way to see a
+solenoid's state is via the small red LED (which indicates if the valve
+is "on" or not), and solenoids can be manually actuated when un-powered
+with the small button adjacent to the LED.
+
+Single acting solenoids apply or vent pressure from a single output
+port. They are typically used either when an external force will provide
+the return action of the cylinder (spring, gravity, separate mechanism)
+or in pairs to act as a double solenoid. A double solenoid switches air
```

Comment:
The photo below describes the 2 kinds of solenoids that have ever been included in the KOP. 

![FRC Solenoids](https://user-images.githubusercontent.com/3799260/61171659-aec6c000-a548-11e9-8f99-d9f5b27d1657.png)

The image on the left is what FRC calls "Single" solenoids; on the right is "Double".  As you can see both devices have 2 output ports (labeled 2 and 4).  The primary difference between single and double solenoids is what happens when they are not powered (Like between auto and teleop or at the end of a match).  Double solenoids will stay in their last commanded state.  Single solenoids will go to their default.

Reply:
Correction.  One other valve was included (I think before 2010).  It was a  VPLE18-M5H-4/2-1/4

![valve symbols_42_mono-stable_2 port designations_en](https://user-images.githubusercontent.com/3799260/61171963-9f497600-a54c-11e9-9f2e-1cf11e7b616a.png)

It is the same as the current single solenoids but the two sides of the system share a single exhaust port.